File: 02-sum/main.py
import random
import time

def main():
    total = 0
    randoms = [random.random() for i in range(1000000)]
    start = time.time()
    for i in range(1000000):
        if randoms[i] > 0.5:
            total += randoms[i]
    elapsed = time.time() - start
    print(total)
    print(elapsed*1000) # python time is in seconds, so we multiply by 1000 to get milliseconds

if __name__ == '__main__':
    main()
    

File: 02-sum/main.go
package main

import (
	"math/rand"
	"time"
)

const (
	Length = 1e9
)

func main() {
	var total float32
	randoms := make([]float32, Length)
	for i := 0; i < Length; i++ {
		randoms[i] = rand.Float32()
	}
	start := time.Now()
	for i := 0; i < Length; i++ {
		if randoms[i] > 0.5 {
			total += randoms[i]
		}
	}
	elapsed := time.Since(start)
	println(total)
	println(elapsed.Milliseconds())
}
